Code/Framework/AzCore/Tests/Settings/SettingsRegistryMergeUtilsTests.cpp

```diff
@@ -751,4 +752,277 @@ tags=tools,renderer,metal)"
+                   R"({ "engine_name": "o3de1", "version": "1.2.3"})",
+                   R"({ "engine_name": "o3de2", "version": "2.3.4"})",
+                },
+
+                // project/project.json
+                R"({ "project_name": "TestProject", "engine":"o3de2==2.3.4"  })",
+
+                // project/user/project.json
+                R"({ "engine":"o3de1==1.2.3" })",
+
+                0 // o3de1
+            },
+
+            // Selects the engine specified by path in project/user/project.json
+            SettingsRegistryFindEngineRootParams{
+                // engine.json files
+                {
+                   R"({ "engine_name": "o3de", "version": "1.2.3"})",
+                   R"({ "engine_name": "o3de", "version": "1.2.3"})",
+                },
+
+                // project/project.json
+                R"({ "project_name": "TestProject", "engine":"o3de"  })",
+
+                // project/user/project.json
+                R"({ "engine_path":"<engine_path1>" })",
+
+                1 // 2nd engine, even though both have same name & version
+            },
+
+            // Fails if invalid engine specified in project/user/project.json
+            SettingsRegistryFindEngineRootParams{
+                // engine.json files
+                {
+                   R"({ "engine_name": "o3de", "version": "1.2.3"})",
+                   R"({ "engine_name": "o3de-other", "version": "1.2.3"})",
+                },
+
+                // project/project.json
+                R"({ "project_name": "TestProject", "engine":"o3de"  })",
+
+                // project/user/project.json
+                R"({ "engine_path":"c:/path/not/found" })",
+
+                -1 // engine used by shared project.json 
+            },
+
+        };
+    }
+
+    class SettingsRegistryMergeUtilsFindEngineRootFixture
+        : public UnitTest::LeakDetectionFixture
+        , public ::testing::WithParamInterface<SettingsRegistryFindEngineRootParams>
+    {
+    public:
+        void SetUp() override
+        {
+            using namespace rapidjson;
+
+            m_registry = AZStd::make_unique<AZ::SettingsRegistryImpl>();
+
+            // Create the manifest json files
+            const auto& params = GetParam();
+            auto tempRootFolder = AZ::IO::FixedMaxPath(m_testFolder.GetDirectory());
+            AZ::IO::FixedMaxPath o3deManifestFilePath = tempRootFolder / "o3de" / "o3de_manifest.json";
+            AZ::IO::FixedMaxPath projectPath = tempRootFolder / "project" ;
+            AZ::IO::FixedMaxPath projectManifestPath = projectPath / "project.json";
+            AZ::IO::FixedMaxPath projectUserPath = tempRootFolder / "project" /  "user";
+            AZ::IO::FixedMaxPath userProjectManifestPath = projectUserPath / "project.json";
+
+            for (int i = 0; i < params.m_engineManifestsJson.size(); ++i)
+            {
+                const AZ::IO::FixedMaxPath enginePath = tempRootFolder / AZStd::string::format("engine%d",i) ;
```

Comment:
nitpick: missing whitespace around comma
```suggestion
                const AZ::IO::FixedMaxPath enginePath = tempRootFolder / AZStd::string::format("engine%d" , i) ;
```